### PR TITLE
Add UnionRectangles extension method.

### PR DESCRIPTION
**Pull request by RoosterDragon**
Add a helper method for this operation which occurs in a few places.

### CODE REVIEW

File: OpenRA.Game/Exts.cs [hunk 1]
```diff
@@ -151,6 +151,26 @@ static T Random<T>(IEnumerable<T> ts, MersenneTwister r, bool throws)
 				return xs.ElementAt(r.Next(xs.Count));
 		}
 
+		public static Rectangle UnionRectangles(this IEnumerable<Rectangle> rects)
```
Comment (pchote) [inline OpenRA.Game/Exts.cs]: The `Rectangles` is implicit from it acting on `IEnumerable<Rectangle>`. Can we call this just `Union` instead to match `Product` and `SymmetricDifference` below?

Comment (RoosterDragon) [reply OpenRA.Game/Exts.cs]: Renamed.

**Review by teinarss — DISMISSED** [review]

File: OpenRA.Mods.Common/Widgets/ActorPreviewWidget.cs [hunk 2]
```diff
@@ -58,15 +58,9 @@ public void SetPreview(ActorInfo actor, TypeDictionary td)
...
-			{
-				var b = r.First();
-				foreach (var rr in r.Skip(1))
-					b = Rectangle.Union(b, rr);
-
-				IdealPreviewSize = new int2(b.Width, b.Height);
-				PreviewOffset = -new int2(b.Left, b.Top) - IdealPreviewSize / 2;
-			}
+			var b = r.Union();
+			IdealPreviewSize = new int2(b.Width, b.Height);
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/ActorPreviewWidget.cs]: Assuming `!r.Any()`, wouldn't `b` then be `Rectangle.Empty` which would then mean setting this to `(0, 0)`?
Can we safely assume there is always at least one entry in `r`?

**Review by abcdefg30 — DISMISSED** [review]
Looks good to me otherwise.

File: OpenRA.Mods.Common/Widgets/ActorPreviewWidget.cs (hunk 2 above)
Comment (RoosterDragon) [reply OpenRA.Mods.Common/Widgets/ActorPreviewWidget.cs]: Good spot here - this is a behaviour change if `!r.Any()`

Thankfully, just a fraction above:
```c#
PreviewOffset = int2.Zero;
IdealPreviewSize = int2.Zero;
```

so even if it is empty (which would be odd) we're already doing what it wanted anyway. I've streamlined the code a touch as a result.

Other changes guarded by a `.Any` look similarly safe to drop that check in.

**Review by abcdefg30 — APPROVED** [review]

**Pull request merged**